src/api.py

```diff
@@ -246,14 +249,49 @@ def get_default_agent(self, throw_if_missing: bool = True) -> Optional[Agent]:
         return sub_agent
 
 
+class GameREPL(AgentREPL):
+    last_seen_block = 0
+
+    def print_object_or_objects(
+        self, output: Union[List, Any], metadata: Optional[Dict[str, Any]] = None
+    ):
+        context = AgentContext.get_or_create(
+            client=self.agent_instance.client,
+            context_keys={"id": "default"},
+            searchable=False,
+        )
+        for block in context.chat_history.file.blocks:
+            if block.index_in_file > self.last_seen_block:
+                if block.stream_state == StreamState.STARTED:
+                    while block.stream_state not in [
+                        StreamState.COMPLETE,
+                        StreamState.ABORTED,
+                    ]:
+                        time.sleep(0.4)
+                        block = Block.get(block.client, _id=block.id)
+            self.print_new_block(block)
+        self.last_seen_block = context.chat_history.file.blocks[-1].index_in_file
+        super().print_object_or_objects(output, metadata)
+
+    def print_new_block(self, block: Block):
+        if TagKind.STATUS_MESSAGE not in [tag.kind for tag in block.tags]:
+            tag_texts = "".join(
+                sorted(set([f"[{tag.kind},{tag.name}]" for tag in block.tags]))
```

Comment:
question(non-blocking): does this produce long sets of tags for certain blocks?

Reply:
Somewhat.  Open to any changes you might like here as a fellow CLI debugger.  I thought it somewhat useful to be able to see the tags emerging as well as the block text as they came out.